src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -456,6 +457,22 @@ private void RenderTagOutput(string tagOutputMethodName)
                    .WriteEndMethodInvocation();
         }
 
+        private void RenderSelfClosing(bool selfClosing)
+        {
+            if (_designTimeMode)
+            {
+                return;
+            }
+            if (selfClosing == false)
+            {
+                return;
+            }
+            _writer.Write(ExecutionContextVariableName)
+                   .Write(".SelfClosing")
```

Comment:
You need a configuration setup for the name `SelfClosing`: `ExecutionContextSelfClosingPropertyName`.

This can be done here: https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor/Generator/GeneratedTagHelperContext.cs